ament_cmake_test/ament_cmake_test/__init__.py

```diff
@@ -185,11 +190,18 @@ def log(msg, **kwargs):
             line = proc.stdout.readline()
             if not line:
                 break
-            decoded_line = line.decode()
+            for i, encoding in enumerate(encodings):
+                try:
+                    decoded_line = line.decode(encoding)
+                except UnicodeDecodeError:
+                    if i == len(encodings) - 1:
+                        raise
+                else:
+                    break
             print(decoded_line, end='')
             output += decoded_line
             if output_handle:
-                output_handle.write(line)
+                output_handle.write(decoded_line.encode())
```

Comment:
Why is this change necessary, isn't `line` already encoded?

Reply:
When using `line` directly it is not ensured that the characters in that string can be encoded into the encoding which is used for `stdout`.

By first successfully decoding it (with one of the two possible encodings) and then re-encoding it is ensure that the string can be written to `stdout` without an exception.